api_examples/validate_receipt_smtp.js

```diff
@@ -0,0 +1,64 @@
+const request = require('superagent');
+const nodemailer = require('nodemailer');
+
+const checkReceived = (receiveAddress, sendAddress, baseURL, headers) => {
+    const apiURL = `${baseURL}/addresses/${receiveAddress}/messages`;
+    request
+        .get(apiURL)
+        .set(headers)
+        .then(res => {
+            res.body.forEach((message) => {
+                if (message.from[0].address == sendAddress) {
+                    console.log(message.received)
+                } else {
+                    console.log(`${receiveAddress} has not received an email from ${sendAddress}`)
+                }
+            })
+        })
+        .catch(err => {
+            console.log(err.message)
+        });
+}
+
+const SMTP_USERNAME = 'mysmtp_user';
+const SMTP_PASSWORD = 'mysmtp_password';
+const SMTP_SERVER = 'mysmtp_server.company.com';
+const SMTP_PORT = 587;
+const FROM_ADDRESS = 'myuser@company.com';
+const FROM_NAME = 'MyTest User';
+const SUBJECT = 'Testing email to Mailsac';
+
+const API_TOKEN = 'MY_API_TOKEN_FROM_MAILSAC';
+const BASE_URL = 'https://mailsac.com/api';
+
+const BODY_TEXT = "Mailsac SMTP validate Email Send\r\nThis email was sent using the SMTP to test receipt of an email.";
+
+for (let i = 1; i < 11; i++) {
+    const transporter = nodemailer.createTransport({
+        host: SMTP_SERVER,
+        port: SMTP_PORT,
+        auth: {
+            user: SMTP_USERNAME,
+            pass: SMTP_PASSWORD,
+        },
+    });
+
+    const result = transporter.sendMail({
```

Comment:
Unfortunately this won't work as intended.

Since JavaScript handles the I/O asyncronously, and calls the callback asynchronously, it won't block the main thread.

It will queue up all 10 emails emails at once then execute them concurrently. Then check for emails 10 times concurrently. It's possible you copied this from somewhere in our examples, if so, sorry about that (though definitely a concept to understand).

Probably the most straightforward approach would be to turn checkReceived and sendMail into promises. psuedocode:

```javascript
const checkReceived = () => {
  // .....
  *** return *** request
         .get(apiURL)
         .set(headers)
         // ...
}

```

```javascript
const sendMailAsync = () => new Promise((resolve, reject) => {
  transporter.sendMail(params, (err) => {
    if (err) { reject(err) } else { resolve() }
  })
})
```

then make the setTimeout asynchronous and `await` `sendMail()`

```javascript
setTimeout((async () => {
  for (let i = 0; i < 10; i++) {
    await checkReceived()
  }
}).catch(err => console.error("something broke", err)))
```

Perhaps @mjmayer knows of a place where we have a better example you could follow.